logger/logrus.go

```diff
@@ -98,24 +122,35 @@ func (l *LogrusEntry) WithField(key string, value interface{}) Entry {
 	return &LogrusEntry{
 		entry: l.entry.WithField(key, value),
 		level: l.level,
+		sampler: l.sampler,
 	}
 }
 
 func (l *LogrusEntry) WithString(key string, value string) Entry {
 	return &LogrusEntry{
 		entry: l.entry.WithField(key, value),
 		level: l.level,
+		sampler: l.sampler,
 	}
 }
 
 func (l *LogrusEntry) WithFields(fields map[string]interface{}) Entry {
 	return &LogrusEntry{
 		entry: l.entry.WithFields(fields),
 		level: l.level,
+		sampler: l.sampler,
 	}
 }
 
 func (l *LogrusEntry) Logf(f string, args ...interface{}) {
+	if l.sampler != nil {
+		rate := l.sampler.GetSampleRate(fmt.Sprintf("%s:%s", l.level, fmt.Sprintf(f, args...)))
+		if shouldDrop(uint(rate)){
+			return 
+		}
+		l.entry.WithField("sample_rate", rate)
```

Comment:
Oh, I'm not familiar with Beelines. Do you mind to explain a bit more on what the incompatibility looks like here?

Reply:
Whenever Honeycomb receives data in a column called SampleRate, we use that in calculations as the sample rate. It's how we compensate for sampling. So we should name the field the way Honeycomb expects to receive it, in case that data is ever forwarded to us.